### PR TITLE
MIGRATE (WIP)

### PR DESCRIPTION
**Pull request by valtandor**
New EasyConfigs for various versions of MIGRATE with various toolchains

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+binary_files = ["migrate-n-mpi", "migrate-n"]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
+sanity_check_paths = {
+    'files': ['bin/%s' % x for x in binary_files],
+    'dirs': []
+}
+
+# You SHOULD change the following line; Kindly consult other easyconfigs for possible options
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: comments like this should be removed

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+prebuildopts = 'cd ./src && '
+
+buildopts = ' mpis && make cleanall && make thread '
+
+preinstallopts = 'cd ./src && mkdir -p %(installdir)s/{bin,man/man1} && '
+
+binary_files = ["migrate-n-mpi", "migrate-n"]
+
+# The sanity test MUST be tuned before going production and submitting your contribution to upstream git repositories
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: comments like this should be removed

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+parallel = 1
+
+prebuildopts = 'cd ./src && '
+
+buildopts = ' mpis && make cleanall && make thread '
+
+preinstallopts = 'cd ./src && mkdir -p %(installdir)s/{bin,man/man1} && '
+
+binary_files = ["migrate-n-mpi", "migrate-n"]
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: only used in one place, so no need to use a local variable?

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
+oldversions="oldversions/%s.x/%s/" % (".".join(version.split('.')[:1]), ".".join(version.split('.')[:2])) 
+sources = ['%(namelower)s-%(version)s.src.tar.gz']
+source_urls = ['http://popgen.sc.fsu.edu/newversions/', 'http://popgen.sc.fsu.edu/currentversions/', 'http://popgen.sc.fsu.edu/%s' % oldversions ]
+
+patches = []
+
+dependencies = [('zlib', '1.2.5')]
+
+preconfigopts = 'cd ./src && sed -i "s/ \-fast / /g" configure && '
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: use `start_dir = 'src'`

I'm not a big fan of using `sed`, but we don't have a better alternative yet (other than creating a patch)

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'} 
+toolchainopts = {'opt': True, 'optarch': True, 'usempi': True, 'unroll': True, 'static': False}
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
+oldversions="oldversions/%s.x/%s/" % (".".join(version.split('.')[:1]), ".".join(version.split('.')[:2])) 
+sources = ['%(namelower)s-%(version)s.src.tar.gz']
+source_urls = ['http://popgen.sc.fsu.edu/newversions/', 'http://popgen.sc.fsu.edu/currentversions/', 'http://popgen.sc.fsu.edu/%s' % oldversions ]
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: please reformat, only URL per (indented) line

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+easyblock = 'ConfigureMake'
+
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
+
+toolchain = {'name': 'goolf', 'version': '1.4.10'} 
+toolchainopts = {'opt': True, 'optarch': True, 'usempi': True, 'unroll': True, 'static': False}
+
+# For sources line to work correctly with --try-software-version, you MUST employ %s OR use a construct like SOURCE_TAR_GZ
+oldversions="oldversions/%s.x/%s/" % (".".join(version.split('.')[:1]), ".".join(version.split('.')[:2]))
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: we support `%(version_major)s` and `%(version_major_minor)s` for this

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 7]
```diff
@@ -0,0 +1,46 @@
+# 
+# This is an easyconfig file for EasyBuild, see https://github.com/hpcugent/easybuild
+# Authors::   Jordi Blasco <jordi.blasco@nesi.org.nz>
+# New Zealand eScience Infrastructure (NeSI)
+
+name = 'MIGRATE'
+version = "3.4.4" 
+easyblock = 'ConfigureMake'
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: `easyblock` line should go above `name` (with an empty line in between)

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 8]
```diff
@@ -0,0 +1,46 @@
+# 
+# This is an easyconfig file for EasyBuild, see https://github.com/hpcugent/easybuild
+# Authors::   Jordi Blasco <jordi.blasco@nesi.org.nz>
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: @valtandor: is this correct?

File: easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+# New Zealand eScience Infrastructure (NeSI)
+
+name = 'MIGRATE'
+version = "3.4.4" 
+easyblock = 'ConfigureMake'
+
+homepage = 'http://popgen.sc.fsu.edu/Migrate/Migrate-n.html'
+description = """Migrate estimates effective population sizes and past migration rates between n population assuming a migration matrix model with asymmetric migration rates and different subpopulation sizes. Migrate uses maximum likelihood or Bayesian inference to jointly estimate all parameters."""
```
Comment (boegel) [inline easybuild/easyconfigs/m/MIGRATE/MIGRATE-3.4.4-goolf-1.4.10.eb]: please break up long lines

**Comment by boegel** [comment]
@valtandor: same remarks apply to all easyconfigs, let us know if they're tackled

**Comment by valtandor** [comment]
@boegel: This branch has easyconfigs in it that were written by colleagues and then deployed on our cluster. I will confer with those colleagues (including@jordiblasco).

**Comment by boegel** [comment]
@valtandor: any updates on handling the remarks?

**Comment by boegel** [comment]
@valtandor:

Irresolvable dependencies encountered: zlib/1.2.5-goolf-1.5.14, zlib/1.2.5-ictce-5.4.0'

**Comment by valtandor** [comment]
@jordiblasco this is my MIGRATE PR

**Comment by valtandor** [comment]
@boegel: After consultation with @jordiblasco I've decided not to pursue this further. If we ever need to install MIGRATE again we will do a fresh PR at the time. Thanks for taking the time to follow up.

**Pull request closed without merging**